server/src/main/scala/org/http4s/server/middleware/Logger.scala

```diff
@@ -15,21 +17,37 @@ import org.log4s.getLogger
 object Logger {
   private[this] val logger = getLogger
 
-  def apply[F[_]: Concurrent](
+  def apply[G[_]: MonadError[?[_], Throwable], F[_]: Concurrent](
       logHeaders: Boolean,
       logBody: Boolean,
+      fk: F ~> G,
       redactHeadersWhen: CaseInsensitiveString => Boolean = Headers.SensitiveHeaders.contains,
       logAction: Option[String => F[Unit]] = None
-  )(@deprecatedName('httpService) http: Kleisli[F, Request[F], Response[F]])
-    : Kleisli[F, Request[F], Response[F]] = {
+  )(@deprecatedName('httpService) http: Http[G, F]): Http[G, F] = {
     val log: String => F[Unit] = logAction.getOrElse { s =>
       Sync[F].delay(logger.info(s))
     }
-    ResponseLogger(logHeaders, logBody, redactHeadersWhen, log.pure[Option])(
-      RequestLogger(logHeaders, logBody, redactHeadersWhen, log.pure[Option])(http)
+    ResponseLogger(logHeaders, logBody, fk, redactHeadersWhen, log.pure[Option])(
+      RequestLogger(logHeaders, logBody, fk, redactHeadersWhen, log.pure[Option])(http)
     )
   }
 
+  def httpApp[F[_]: Concurrent](
```

Comment:
We haven't specialized our others that need an `~>` like this.  I think we should do all or none.  Not strongly opinionated which.